extensions/interactions/FractionInput/directives/FractionInputValidationService.js

```diff
@@ -107,6 +109,13 @@ oppia.factory('FractionInputValidationService', [
               lbi: false,
               ubi: false,
             };
+
+            var denominator = {
```

Comment:
matchedDenominator (odd to have denominator.denominator; try and treat "denominator + reference" as a separate construct)

Reply:
I changed the name. Not sure what the second part of your comment meant.

Reply:
Changes look good. What I'm saying is that it's odd to call two things the "denominator". It's much better for the two concepts ("denominator with references", vs the raw denominator) to have separate names.

Reply:
ok.. So basically object.object is ambiguous and keep away from that... Right?

Reply:
I guess that's another way to say it. But my original point was that you really have two different conceptual ideas here, so it's better to name them different things in order to avoid confusion.

Reply:
Hmmm... Ok I'll keep that in mind :)